packages/salesforcedx-apex-replay-debugger/src/adapter/apexReplayDebug.ts

```diff
@@ -289,10 +324,19 @@ export class ApexReplayDebug extends LoggingDebugSession {
     this.sendResponse(response);
   }
 
-  protected async scopesRequest(
+  public async scopesRequest(
     response: DebugProtocol.ScopesResponse,
     args: DebugProtocol.ScopesArguments
   ): Promise<void> {
+    try {
+      if (this.logContext.hasHeapDumpForTopFrame()) {
+        this.logContext.copyStateForHeapDump();
+        this.replaceVariablesWithHeapDump();
+      }
+    } catch (error) {
+      this.warnToDebugConsole(nls.localize('reconcile_heapdump_error', error));
```

Comment:
Can we get some specific information like the log line number added to the error? I realize that they're probably either stepping or running to a breakpoint but this seems like at least outputting the log line number would be good here.